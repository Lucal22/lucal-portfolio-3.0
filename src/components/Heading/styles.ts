import styled, { css, DefaultTheme } from 'styled-components';

export type HeadingStyleProps = {
  color?: 'white' | 'black';
  size?: 'small' | 'medium' | 'big' | 'xBig' | 'large' | null;
};

const fontSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.size.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.size.medium};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.size.big};
  `,
  xBig: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.size.xBig};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.size.large};
  `,
};

export const Heading = styled.h1<HeadingStyleProps>`
  ${({ theme, color, size }) => css`
    color: ${color};
    ${size ? fontSize[size](theme) : null}
  `}
`;
