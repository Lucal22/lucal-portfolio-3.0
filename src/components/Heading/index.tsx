import React from 'react';
import * as Styled from './styles';
import { HeadingStyleProps } from './styles';

export type HeadingProps = HeadingStyleProps & {
  as?: 'h1' | 'h2' | 'h3';
  children: React.ReactNode;
};

export default function Heading({
  as = 'h1',
  children,
  color = 'white',
  size = 'big',
}: HeadingProps) {
  return (
    <Styled.Heading size={size} color={color} as={as}>
      {children}
    </Styled.Heading>
  );
}
