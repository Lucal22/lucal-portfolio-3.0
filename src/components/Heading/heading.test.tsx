import { screen } from '@testing-library/react';
import { renderTheme } from '../../utils/renderTheme';
import Heading from './index';

describe('<Heading />', () => {
  it('should render with default values', () => {
    const heading = renderTheme(<Heading>Olá mundo</Heading>);
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(heading).toMatchSnapshot();
  });

  it('should render with small size', () => {
    const heading = renderTheme(
      <Heading color={'white'} size={'small'} as={'h1'}>
        Olá mundo
      </Heading>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(heading).toMatchSnapshot();
  });

  it('should render with medium size', () => {
    const heading = renderTheme(
      <Heading color={'black'} size={'medium'} as={'h2'}>
        Olá mundo
      </Heading>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(heading).toMatchSnapshot();
  });

  it('should render with big size', () => {
    const heading = renderTheme(
      <Heading color={'black'} size={'big'} as={'h3'}>
        Olá mundo
      </Heading>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(heading).toMatchSnapshot();
  });

  it('should render with xBig size', () => {
    const heading = renderTheme(
      <Heading color={'black'} size={'xBig'} as={'h1'}>
        Olá mundo
      </Heading>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(heading).toMatchSnapshot();
  });

  it('should render with large size', () => {
    const heading = renderTheme(
      <Heading color={'black'} size={'large'} as={'h2'}>
        Olá mundo
      </Heading>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(heading).toMatchSnapshot();
  });

  it('should render with size null', () => {
    const heading = renderTheme(
      <Heading color={'black'} size={null} as={'h2'}>
        Olá mundo
      </Heading>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(heading).toMatchSnapshot();
  });
});
