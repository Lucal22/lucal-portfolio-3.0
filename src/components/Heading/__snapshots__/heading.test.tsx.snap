// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render with big size 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  color: black;
  font-size: 2.4rem;
}

<body>
    <div>
      <h3
        class="c0"
        color="black"
      >
        Olá mundo
      </h3>
    </div>
  </body>,
  "container": .c0 {
  color: black;
  font-size: 2.4rem;
}

<div>
    <h3
      class="c0"
      color="black"
    >
      Olá mundo
    </h3>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Heading /> should render with default values 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  color: white;
  font-size: 2.4rem;
}

<body>
    <div>
      <h1
        class="c0"
        color="white"
      >
        Olá mundo
      </h1>
    </div>
  </body>,
  "container": .c0 {
  color: white;
  font-size: 2.4rem;
}

<div>
    <h1
      class="c0"
      color="white"
    >
      Olá mundo
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Heading /> should render with large size 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  color: black;
  font-size: 4rem;
}

<body>
    <div>
      <h2
        class="c0"
        color="black"
      >
        Olá mundo
      </h2>
    </div>
  </body>,
  "container": .c0 {
  color: black;
  font-size: 4rem;
}

<div>
    <h2
      class="c0"
      color="black"
    >
      Olá mundo
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Heading /> should render with medium size 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  color: black;
  font-size: 1.8rem;
}

<body>
    <div>
      <h2
        class="c0"
        color="black"
      >
        Olá mundo
      </h2>
    </div>
  </body>,
  "container": .c0 {
  color: black;
  font-size: 1.8rem;
}

<div>
    <h2
      class="c0"
      color="black"
    >
      Olá mundo
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Heading /> should render with size null 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  color: black;
}

<body>
    <div>
      <h2
        class="c0"
        color="black"
      >
        Olá mundo
      </h2>
    </div>
  </body>,
  "container": .c0 {
  color: black;
}

<div>
    <h2
      class="c0"
      color="black"
    >
      Olá mundo
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Heading /> should render with small size 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  color: white;
  font-size: 1.6rem;
}

<body>
    <div>
      <h1
        class="c0"
        color="white"
      >
        Olá mundo
      </h1>
    </div>
  </body>,
  "container": .c0 {
  color: white;
  font-size: 1.6rem;
}

<div>
    <h1
      class="c0"
      color="white"
    >
      Olá mundo
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Heading /> should render with xBig size 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  color: black;
  font-size: 3.2rem;
}

<body>
    <div>
      <h1
        class="c0"
        color="black"
      >
        Olá mundo
      </h1>
    </div>
  </body>,
  "container": .c0 {
  color: black;
  font-size: 3.2rem;
}

<div>
    <h1
      class="c0"
      color="black"
    >
      Olá mundo
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
